xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -587,6 +593,17 @@ where
         // and roll the transaction back, so we can fetch updates from the server before
         // being ready to process the message for a second time.
         let mut processed_message = None;
+        if is_test_mode_aead_msg() {
```

Comment:
@antews @cameronvoell we need to mimic the failure somehow for tests, I can extract the whole test section to another function for tests but like we did here https://github.com/xmtp/libxmtp/blob/4ab7ed92d8c3526a66d8ed5156417bde0495bbad/xmtp_mls/src/groups/mls_sync.rs#L1974

Reply:
https://github.com/xmtp/libxmtp/issues/1903